expression/typeinfer_test.go

```diff
@@ -2976,6 +2975,14 @@ func (s *InferTypeSuite) createTestCase4JSONFuncs() []typeInferTestCase {
 		{"json_merge(c_json, c_json)", mysql.TypeJSON, charset.CharsetUTF8MB4, mysql.BinaryFlag, mysql.MaxBlobWidth, 0},
 		{"json_object('k', 'v')", mysql.TypeJSON, charset.CharsetUTF8MB4, mysql.BinaryFlag, mysql.MaxBlobWidth, 0},
 		{"json_array('k', 'v')", mysql.TypeJSON, charset.CharsetUTF8MB4, mysql.BinaryFlag, mysql.MaxBlobWidth, 0},
+		{"json_pretty(c_json)", mysql.TypeLongBlob, charset.CharsetUTF8MB4, mysql.BinaryFlag, mysql.MaxLongBlobWidth, types.UnspecifiedLength},
```

Comment:
It seems that it only has half of the max length on mysql8:

```
Type:       LONG_BLOB
Collation:  utf8mb4_general_ci (45)
Length:     268435456
```

Reply:
it is actually `16777216 * 4 * 4`, `16777216` is the max blob length, `*4` for `utf8mb4` max length, and another `*4` for `utf8mb4` to `binary`. It is hard to compatible with MySQL now, I think it is ok to set it to `16777216 * 4`